apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -459,39 +457,26 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
     Config.device_heartbeat_interval_ms!() |> div(1000)
   end
 
-  defp set_connection_info!(db_client, device_id, timestamp_ms, ip_address) do
-    device_update_statement = """
-    UPDATE devices
-    SET last_connection=:last_connection, last_seen_ip=:last_seen_ip
-    WHERE device_id=:device_id
-    """
+  defp set_connection_info!(realm, device_id, timestamp, ip_address) do
+    keyspace_name = Realm.keyspace_name(realm)
 
-    device_update_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(device_update_statement)
-      |> DatabaseQuery.put(:device_id, device_id)
-      |> DatabaseQuery.put(:last_connection, timestamp_ms)
-      |> DatabaseQuery.put(:last_seen_ip, ip_address)
-      |> DatabaseQuery.consistency(:local_quorum)
+    changes = %{
+      last_connection: timestamp,
+      last_seen_ip: ip_address
+    }
 
-    DatabaseQuery.call!(db_client, device_update_query)
+    changeset = compute_device_update_changes(device_id, changes)
+
+    Repo.update!(changeset, prefix: keyspace_name, consistency: :local_quorum)
   end
 
-  defp refresh_device_connected!(db_client, device_id, ttl) do
-    refresh_connected_statement = """
-    UPDATE devices
-    USING TTL #{ttl}
-    SET connected=true
-    WHERE device_id=:device_id
-    """
+  defp refresh_device_connected!(realm, device_id, ttl) do
+    keyspace_name = Realm.keyspace_name(realm)
 
-    refresh_connected_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(refresh_connected_statement)
-      |> DatabaseQuery.put(:device_id, device_id)
-      |> DatabaseQuery.consistency(:local_quorum)
+    changeset = compute_device_update_changes(device_id, %{connected: true})
 
-    DatabaseQuery.call!(db_client, refresh_connected_query)
+    # We use `insert` here becuase Exandra does not support ttl on updates. However, this is an upsert in Scylla.
+    Repo.insert!(changeset, prefix: keyspace_name, ttl: ttl, consistency: :local_quorum)
```

Comment:
It's fine to use an insert instead of an update, but I would double check which data is being written by Ecto: it should not overwrite all fields of an existing device with `nil` just because we're building a shallow Ecto struct here with just two fields.